src/login/pam_systemd.c

```diff
@@ -47,20 +49,79 @@
 
 #define LOGIN_SLOW_BUS_CALL_TIMEOUT_USEC (2*USEC_PER_MINUTE)
 
+static int parse_caps(
+                pam_handle_t *handle,
+                const char *value,
+                uint64_t *caps) {
+
+        bool subtract;
+        int r;
+
+        assert(handle);
+        assert(value);
+
```

Comment:
if `caps` is NULL this function will never do anything, no? just short-cut it here then?